### PR TITLE
Initial version of the SDC lib codelab

### PR DESCRIPTION
**Pull request by jingtang10**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

**Description**
This is initially developed as part of the digital square webinar: https://github.com/openhie/FoodAllergy-webinar/tree/master/android

**Alternative(s) considered**
NA

**Type**
Documentation

**Screenshots (if applicable)**

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://github.com/google/android-fhir/wiki/Contributing) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

File: codelabs/datacapture/.gitignore [hunk 1]
```diff
@@ -0,0 +1,15 @@
+*.iml
```
Comment (omarismail94) [inline codelabs/datacapture/.gitignore:1]: Can all these files not be added in the root `.gitignore`?

File: codelabs/datacapture/app/build.gradle [hunk 2]
```diff
@@ -0,0 +1,48 @@
+plugins {
```
Comment (omarismail94) [inline codelabs/datacapture/app/build.gradle:1]: Is there a reason the file format is not `.kts` like the rest of the files?

File: codelabs/datacapture/app/build.gradle [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        sourceCompatibility JavaVersion.VERSION_1_8
+        targetCompatibility JavaVersion.VERSION_1_8
+    }
+    kotlinOptions {
+        jvmTarget = '1.8'
+    }
+}
+
+dependencies {
+    implementation 'androidx.core:core-ktx:1.7.0'
```
Comment (omarismail94) [inline codelabs/datacapture/app/build.gradle:35]: These libs are in the `Dependencies` class IIRC. Should we reference them via that class?

File: codelabs/datacapture/app/proguard-rules.pro [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+#   public *;
+#}
+
+# Uncomment this to preserve the line number information for
+# debugging stack traces.
+#-keepattributes SourceFile,LineNumberTable
+
+# If you keep the line number information, uncomment this to
+# hide the original source file name.
+#-renamesourcefileattribute SourceFile
```
Comment (omarismail94) [inline codelabs/datacapture/app/proguard-rules.pro:21]: nit: add empty new line below

File: codelabs/datacapture/app/src/main/AndroidManifest.xml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+            android:exported="true">
+            <intent-filter>
+                <action android:name="android.intent.action.MAIN" />
+
+                <category android:name="android.intent.category.LAUNCHER" />
+            </intent-filter>
+        </activity>
+    </application>
+
+</manifest>
```
Comment (omarismail94) [inline codelabs/datacapture/app/src/main/AndroidManifest.xml:23]: nit: add empty new line below

File: codelabs/datacapture/app/build.gradle (hunk 2 above)
Comment (jingtang10) [reply codelabs/datacapture/app/build.gradle]: done

**Review by jingtang10 — COMMENTED** [review]
@omarismail94 this is done - ptal! thanks

File: codelabs/datacapture/app/build.gradle.kts [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+    }
+
+    buildTypes {
+        release {
+            isMinifyEnabled = false
+            proguardFiles(getDefaultProguardFile("proguard-android-optimize.txt"), "proguard-rules.pro")
+        }
+    }
+    compileOptions {
+        sourceCompatibility = JavaVersion.VERSION_1_8
```
Comment (omarismail94) [inline codelabs/datacapture/app/build.gradle.kts:26]: Is there any reason we dont use Java 11 as the base? I am assume it has to do with old Android versions - is that the case?

File: codelabs/datacapture/app/src/main/java/com/example/foodallergy/MainActivity.kt [hunk 7]
```diff
@@ -1,30 +1,46 @@
+/*
+ * Copyright 2021 Google LLC
```
Comment (omarismail94) [inline codelabs/datacapture/app/src/main/java/com/example/foodallergy/MainActivity.kt:2]: nit: update header generator to use 2022

**Review by omarismail94 — APPROVED** [review]
LGTM with just a couple of nits! Nice changes btw

File: codelabs/datacapture/app/build.gradle.kts (hunk 6 above)
Comment (jingtang10) [reply codelabs/datacapture/app/build.gradle.kts:26]: We have been using java8 for the main project. There just isn't a need to upgrade to java11 yet. I think java8 support was only introduced in Android studio recently. See "Use Java 8 language features and APIs  |  Android Developers" https://developer.android.com/studio/write/java8-support

 I suggest we address this issue separately.

File: codelabs/datacapture/app/src/main/java/com/example/foodallergy/MainActivity.kt (hunk 7 above)
Comment (jingtang10) [reply codelabs/datacapture/app/src/main/java/com/example/foodallergy/MainActivity.kt:2]: I'll update spotless the whole project.

Comment (jingtang10) [reply codelabs/datacapture/app/src/main/java/com/example/foodallergy/MainActivity.kt:2]: https://github.com/google/android-fhir/pull/1437

**Pull request merged**